src/components/TextField/TextArea.js

```diff
@@ -0,0 +1,279 @@
+import React, { Component } from 'react';
+import autosize from 'autosize';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import ListPropType from '../../prop-types/list';
+import StyleObjectPropType from '../../prop-types/style';
+import Tooltip from '../Tooltip/Tooltip';
+import TooltipBox from '../Tooltip/TooltipBox';
+import { TooltipBoxStatus, TooltipPosition } from '../Tooltip/TooltipEnums';
+import IconRequired from '../Icon/IconRequired';
+import { TextAreaAutoHeight } from './TextFieldEnums';
+import './TextArea.scss';
+
+const propTypes = {
+    className: PropTypes.string,
+    componentRef: PropTypes.func,
+    hasAutoHeight: ListPropType([
+        TextAreaAutoHeight.ENABLED,
+        TextAreaAutoHeight.DISABLED,
+    ]),
+    hasLabelAlways: PropTypes.bool,
+    id: PropTypes.string,
+    isDisabled: PropTypes.bool,
+    isFullWidth: PropTypes.bool,
+    isReadOnly: PropTypes.bool,
+    isRequired: PropTypes.bool,
+    isValid: PropTypes.bool,
+    label: PropTypes.string,
+    onBlur: PropTypes.func,
+    onChange: PropTypes.func,
+    onEnterKey: PropTypes.func,
+    onFocus: PropTypes.func,
+    onKeyDown: PropTypes.func,
+    onKeyPress: PropTypes.func,
+    onKeyUp: PropTypes.func,
+    placeholder: PropTypes.string,
+    rows: PropTypes.number,
+    style: StyleObjectPropType(),
+    tooltipError: PropTypes.oneOfType([
+        PropTypes.element,
+        PropTypes.string,
+    ]),
+    tooltipHint: PropTypes.oneOfType([
+        PropTypes.element,
+        PropTypes.string,
+    ]),
+    tooltipRequired: PropTypes.oneOfType([
+        PropTypes.element,
+        PropTypes.string,
+    ]),
+    value: PropTypes.string,
+};
+
+const defaultProps = {
+    className: null,
+    componentRef: null,
+    hasAutoHeight: TextAreaAutoHeight.DISABLED,
+    hasLabelAlways: false,
```

Comment:
I thought that Antoine wanted this to be true by default. But perhaps I misunderstood, we should double check.
If it's always visible then the prop should have a different name.

Reply:
Very good point... will check